src/layertreemodel.cpp

```diff
@@ -90,10 +90,16 @@ QVariant LayerTreeModel::data( const QModelIndex& index, int role ) const
 
     case Visible:
     {
-      QgsLayerTreeNode* node = mLayerTreeModel->index2node( mapToSource( index ) );
-      return node->isVisible();
+      if ( QgsLayerTreeModelLegendNode *sym = mLayerTreeModel->index2legendNode( mapToSource( index ) ) )
```

Comment:
I think this will always return true, no?
Don't you have to do:

```
QgsLayerTreeModelLegendNode *sym = mLayerTreeModel->index2legendNode( mapToSource( index ) );
if ( sym )
```

Reply:
No, this makes no difference. It returns false if sym is null in both cases.

But wouldn't mind to change it, if it's ugly or mistakable.

Reply:
interesting!
I found it confusing though.